app/controllers/admin_controller.rb

```diff
@@ -0,0 +1,16 @@
+class AdminController < ApplicationController
+
+  before_action :require_admin
+
+  def index
+  end
+
+  private
+
+  def require_admin
+    return if current_user && current_user.admin
+    flash[:error] = 'Entrance permited. Admin require'
```

Comment:
You can also write flash alert as an argument for redirect_to and that would be best solution